ckanext/datastore/logic/schema.py

```diff
@@ -169,6 +169,20 @@ def datastore_delete_schema() -> Schema:
     return schema
 
 
+def datastore_records_delete_schema() -> Schema:
+    schema = {
+        'resource_id': [not_missing, not_empty, unicode_safe],
+        'force': [ignore_missing, boolean_validator],
+        'filters': [not_missing, json_validator],
```

Comment:
filters must be `{str:str or list of str, ...}` dictionaries, not JSON. We could use a new validator for this.

The filters parameter isn't properly validated or covered in the datastore documentation either. Would be really nice to have a section on filters for the endpoints that support them, kind of like we have for fields: https://docs.ckan.org/en/2.9/maintaining/datastore.html?highlight=datastore#fields

Reply:
@wardi Okay we are now using `dict_only`; I also added `dict_only` and `ignore_missing` to the `datastore_delete` schema.

Reply:
`datastore_delete` also accepts `null` to its filters value, so you might not want to change its validation (or at least we should discuss that change on a separate issue)